### PR TITLE
update geth version used by ci

### PR DESCRIPTION
**Pull request by roberto-bayardo**
**Description**

Update the geth version used by the CI pipeline to support some needed 4844 / Dencun related changes.

**Tests**

**Additional context**

**Metadata**

### CODE REVIEW

**Review by sebastianst — APPROVED** [review]
Now please bump the `ci-builder` image to `v0.46.0-rc.1` in `.circleci/config.yml` line 6.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1158,11 +1158,11 @@ jobs:
       - run:
           name: Install Geth
           command: |
-            wget https://gethstore.blob.core.windows.net/builds/geth-alltools-linux-amd64-1.13.4-3f907d6a.tar.gz
```
Comment (tynes) [inline .circleci/config.yml:1161]: This is faster than `go install`?

Comment (roberto-bayardo) [reply .circleci/config.yml:1161]: i don't know why it was done this way.

Comment (sebastianst) [reply .circleci/config.yml:1161]: I think it's faster, it's the pre-built package from https://geth.ethereum.org/downloads

Comment (sebastianst) [reply .circleci/config.yml:1161]: Interestingly, in the ci-builder image itself, we use `go install`

**Review by Inphi — APPROVED** [review]
LGTM

**Pull request merged**